tests/hil/tests/stream/test.c

```diff
@@ -0,0 +1,107 @@
+#include <golioth/client.h>
+#include <golioth/golioth_debug.h>
+#include <golioth/golioth_sys.h>
+#include <golioth/stream.h>
+#include <string.h>
+
+LOG_TAG_DEFINE(test_stream);
+
+static golioth_sys_sem_t connected_sem;
+
+static void on_client_event(struct golioth_client *client,
+                            enum golioth_client_event event,
+                            void *arg)
+{
+    if (event == GOLIOTH_CLIENT_EVENT_CONNECTED)
+    {
+        golioth_sys_sem_give(connected_sem);
+    }
+}
+
+#define long_json \
```

Comment:
It would be cleaner to store this in a text file and a header file generated by `xxd -i`, checking in both. Then the python script can directly calculate the hash of the txt file. This way, the code is easier to read (no super long array in the source), and there are fewer hardcoded things since both the C and python are generating their data from the same source (the txt file).

Reply:
Great suggestion, thanks. I've updated to use this approach.